org.eclipse.m2e.core.ui/src/org/eclipse/m2e/core/ui/internal/search/util/SearchResultAdaptableFactory.java

```diff
@@ -0,0 +1,128 @@
+/*******************************************************************************
+ * Copyright (c) 2010 Sonatype, Inc.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ *
+ * Contributors:
+ *      Sonatype, Inc. - initial API and implementation
+ *******************************************************************************/
+package org.eclipse.m2e.core.ui.internal.search.util;
+
+import org.eclipse.core.runtime.CoreException;
+import org.eclipse.core.runtime.IAdapterFactory;
+import org.eclipse.core.runtime.IProgressMonitor;
+import org.eclipse.core.runtime.NullProgressMonitor;
+import org.eclipse.core.runtime.OperationCanceledException;
+import org.eclipse.core.runtime.jobs.ISchedulingRule;
+import org.eclipse.jface.resource.ImageDescriptor;
+import org.eclipse.ui.progress.IDeferredWorkbenchAdapter;
+import org.eclipse.ui.progress.IElementCollector;
+import org.eclipse.ui.statushandlers.StatusManager;
+
+import org.eclipse.m2e.core.search.ISearchResultGA;
+import org.eclipse.m2e.core.ui.internal.M2EUIPluginActivator;
+
+
+/**
+ * SearchResultAdaptableFactory
+ *
+ * @author Matthew Piggott
+ */
+public class SearchResultAdaptableFactory implements IAdapterFactory {
```

Comment:
What's the purpose of this adapter? Couldn't we achieve similar results with a more explicit API?

Reply:
> What's the purpose of this adapter? Couldn't we achieve similar results with a more explicit API?

Its been long enough that I don't really know - I recall doing some digging into the the available options for displaying trees and chose what seemed like the most appropriate given the use case.